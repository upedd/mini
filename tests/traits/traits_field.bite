trait A {
    a;

    display() {
        print(a);
    }
}

trait B {
    using A;

    get a {
        print("B.a");
        return "B";
    }
}

class C {
    using A;

    get a {
        print("C.a");
        return "C";
    }
}

class D {
    using A;
    a = "D";
}

class E {
    using B;
}

let c = C();
c.display();
let d = D();
d.display();
let e = E();
e.display();