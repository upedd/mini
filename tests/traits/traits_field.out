C.a
C
D
B.a
B